charts/huawei-csi-plugin/templates/rbac.yaml

```diff
@@ -0,0 +1,314 @@
+    verbs: ["get", "list", "watch"]
+  - apiGroups: [""]
+    resources: ["persistentvolumeclaims/status"]
+    verbs: ["update", "patch"]
+  - apiGroups: [""]
+    resources: ["events"]
+    verbs: ["list", "watch", "create", "update", "patch"]
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: huawei-csi-resizer-role
+subjects:
+  - kind: ServiceAccount
+    name: {{ .Values.serviceAccount.controllerName }}
+    namespace: {{ .Release.Namespace }}
+roleRef:
+  kind: ClusterRole
+  name: huawei-csi-resizer-runner
+  apiGroup: rbac.authorization.k8s.io
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: huawei-csi-resizer-cfg
+rules:
+  - apiGroups: ["coordination.k8s.io"]
+    resources: ["leases"]
+    verbs: ["get", "watch", "list", "delete", "update", "create"]
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: huawei-csi-resizer-role-cfg
+subjects:
+  - kind: ServiceAccount
+    name: {{ .Values.serviceAccount.controllerName }}
+    namespace: {{ .Release.Namespace }}
+roleRef:
+  kind: Role
+  name: huawei-csi-resizer-cfg
+  apiGroup: rbac.authorization.k8s.io
+{{- end }}
+
+{{- if .Values.features.snapshotting.enabled }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: huawei-csi-snapshotter-runner
+rules:
+  - apiGroups: [""]
+    resources: ["events"]
+    verbs: ["list", "watch", "create", "update", "patch"]
+  - apiGroups: ["snapshot.storage.k8s.io"]
+    resources: ["volumesnapshotclasses"]
+    verbs: ["get", "list", "watch"]
+  - apiGroups: ["snapshot.storage.k8s.io"]
+    resources: ["volumesnapshotcontents"]
+    verbs: ["create", "get", "list", "watch", "update", "delete"]
+  - apiGroups: ["snapshot.storage.k8s.io"]
+    resources: ["volumesnapshotcontents/status"]
+    verbs: ["update"]
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: huawei-csi-snapshotter-role
+subjects:
+  - kind: ServiceAccount
+    name: {{ .Values.serviceAccount.controllerName }}
+    namespace: {{ .Release.Namespace }}
+roleRef:
+  kind: ClusterRole
+  name: huawei-csi-snapshotter-runner
+  apiGroup: rbac.authorization.k8s.io
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: huawei-csi-snapshotter-leaderelection
+rules:
+  - apiGroups: ["coordination.k8s.io"]
+    resources: ["leases"]
+    verbs: ["get", "watch", "list", "delete", "update", "create"]
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: huawei-csi-snapshotter-leaderelection
+subjects:
+  - kind: ServiceAccount
+    name: {{ .Values.serviceAccount.controllerName }}
+    namespace: {{ .Release.Namespace }}
+roleRef:
+  kind: Role
+  name: huawei-csi-snapshotter-leaderelection
+  apiGroup: rbac.authorization.k8s.io
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: snapshot-controller-runner
```

Comment:
should be templated and not hardcoded. This is an example why templating with a prefix is always a good option. how do we now know it's regarding huawei-csi if it isn't coming up in the name?

Reply:
done :white_check_mark: